internal/services/kusto/kusto_cluster_resource_test.go

```diff
@@ -238,6 +238,19 @@ func TestAccKustoCluster_vnet(t *testing.T) {
 				check.That(data.ResourceName).Key("virtual_network_configuration.0.subnet_id").Exists(),
 				check.That(data.ResourceName).Key("virtual_network_configuration.0.engine_public_ip_id").Exists(),
 				check.That(data.ResourceName).Key("virtual_network_configuration.0.data_management_public_ip_id").Exists(),
+				check.That(data.ResourceName).Key("virtual_network_configuration_enabled").HasValue("true"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.vnetUpdate(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("virtual_network_configuration.#").HasValue("1"),
+				check.That(data.ResourceName).Key("virtual_network_configuration.0.subnet_id").Exists(),
+				check.That(data.ResourceName).Key("virtual_network_configuration.0.engine_public_ip_id").Exists(),
+				check.That(data.ResourceName).Key("virtual_network_configuration.0.data_management_public_ip_id").Exists(),
+				check.That(data.ResourceName).Key("virtual_network_configuration_enabled").HasValue("false"),
```

Comment:
Can add an addition test step here where we validate that the provider will throw an error when the value is set to `false` and is then attempted to be set to `true`, which isn't allowed?